src/Facility.Definition/ServiceElementInfo.cs

```diff
@@ -63,7 +63,7 @@ private protected ServiceElementInfo(IEnumerable<ServicePart> parts)
 			m_validationErrors = new List<ServiceDefinitionError>();
 		}
 
-		private protected void AddValidationError(ServiceDefinitionError error)
+		public void AddValidationError(ServiceDefinitionError error)
```

Comment:
This pokes some holes in the way that validation can happen, but I don't see anyway around it since fields can't be entirely responsible for their `[validate]` validation.